providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -42,7 +44,15 @@ public FlagdProvider() {
      * @param options {@link FlagdOptions} with
      */
     public FlagdProvider(final FlagdOptions options) {
-        this.flagResolver = new GrpcResolution(options, CacheFactory.getCache(options), this::getState, this::setState);
+        switch (options.getResolverType()) {
+            case inProcess:
+                this.flagResolver = new InProcessResolver(options, this::setState);
+                break;
+            case flagd:
+            default:
```

Comment:
I would say the default path should return an error: invalid resolved type

Reply:
we should never fail in the library IMHO. However, a log line would be nice 🤔

Reply:
here it's done nicely https://github.com/open-feature/java-sdk-contrib/pull/412/files#diff-27f34c6acd23d899adf4c27d00f4d5b9e730090bb2a88efc9a286f3da0bd9dccR58

Reply:
I will add a log line :)

Reply:
ohh actually we can throw an error given this is a configuration step. OF Spec allows this behavior.